minidump-processor/src/op_analysis.rs

```diff
@@ -154,222 +234,341 @@ mod amd64 {
-            Ok(memory_accesses)
+        /// is invalid.
+        fn from_instruction(
+            instruction: Instruction,
+            context: &MinidumpContext,
+        ) -> Result<Vec<Self>, OpAnalysisError> {
+            let mut accesses = Vec::new();
+            if let Some(opcode) = CommonOpcode::from_amd64_opcode(instruction.opcode()) {
+                Self::add_common_opcode_accesses(&mut accesses, opcode, instruction, context)?;
+            } else {
+                Self::add_uncommon_opcode_accesses(&mut accesses, instruction, context)?;
+            }
+            Ok(accesses)
         }
 
-        fn direct(
-            &self,
+        fn add_common_opcode_accesses(
             accesses: &mut Vec<MemoryAccess>,
-            decoded_instruction: &Instruction,
+            opcode: CommonOpcode,
+            instruction: Instruction,
+            context: &MinidumpContext,
         ) -> Result<(), OpAnalysisError> {
-            // Shortcut -- If the instruction doesn't access memory, just return an empty list
-            let mem_size = match decoded_instruction.mem_size() {
+            // Shortcut -- If the instruction doesn't access memory, just return
+            let mem_size = match instruction.mem_size() {
                 Some(access) => access.bytes_size(),
                 None => return Ok(()),
             };
 
-            for idx in 0..decoded_instruction.operand_count() {
-                let operand = decoded_instruction.operand(idx);
+            for idx in 0..instruction.operand_count() {
+                Self::add_common_opcode_explicit_access(
+                    accesses,
+                    opcode,
+                    instruction.operand(idx),
+                    idx,
+                    mem_size,
+                    context,
+                )?;
+            }
 
-                let maybe_access = match operand {
-                    Operand::DisplacementU32(disp) => Some(MemoryAccess {
-                        address: disp.into(),
-                        is_likely_null_pointer_dereference: false,
-                        is_likely_guard_page: false,
-                        size: mem_size,
-                    }),
-                    Operand::DisplacementU64(disp) => Some(MemoryAccess {
-                        address: disp,
-                        is_likely_null_pointer_dereference: false,
-                        is_likely_guard_page: false,
-                        size: mem_size,
-                    }),
-                    other_operand => {
-                        if let Some(op_info) = RegOperandInfo::try_from_operand(other_operand) {
-                            Some(self.calculate_address(mem_size, op_info)?)
-                        } else {
-                            None
-                        }
-                    }
-                };
+            Self::add_common_opcode_implicit_access(accesses, opcode, mem_size, context)?;
```

Comment:
Assuming that Yaxpeax doesn't account for implicit accesses, that means that `mem_size` is not useful here, and you will have to determine the memory access size based on the opcode (I think almost all of them will be 8 bytes).